### PR TITLE
feat(app): range error handling for number RTPs

### PR DESCRIPTION
**Pull request by ncdiehl11**
closes [AUTH-104](https://opentrons.atlassian.net/browse/AUTH-104)

# Overview

To build initial error handling for desktop RTP, I add a range check to ensure that the input value for a given int/float type RTP falls within the user-defined range for that RTP.

# Test Plan

- from protocol card/protocol details page, begin protocol setup
- select a robot and proceed to parameters form
- for number-type RTPs, enter an invalid value (falling outside of the defined min-max range for that parameter)
- observe error message
Screenshot 2024-04-01 at 2 19 36 PM

- for number-type RTPs, delete input field content such that value is `NaN`
- observe error message
Screenshot 2024-04-01 at 12 06 34 PM

# Changelog

- add basic error handling on change to mapped RTP in `ChooseRobotSlideout`
- add tests

# Review requests

authorship devs

# Risk assessment

low

[AUTH-104]: https://opentrons.atlassian.net/browse/AUTH-104?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx [hunk 1]
```diff
@@ -234,6 +234,33 @@ describe('ChooseRobotSlideout', () => {
...
 
+  it('renders error message for runtime parameter out of range', () => {
+    render({
+      onCloseClick: vi.fn(),
+      isExpanded: true,
+      isSelectedRobotOnDifferentSoftwareVersion: false,
+      selectedRobot: null,
+      setSelectedRobot: mockSetSelectedRobot,
+      title: 'choose robot slideout title',
+      robotType: 'OT-2 Standard',
```
Comment (jerader) [inline app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: nit: can you use the const? `OT2_ROBOT_TYPE`?

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx [hunk 2]
```diff
@@ -234,6 +234,33 @@ describe('ChooseRobotSlideout', () => {
...
+        {
+          value: 1000,
+          displayName: 'EtoH Volume',
+          variableName: 'ETOH_VOLUME',
+          description: '70% ethanol volume',
+          type: 'float',
+          suffix: 'mL',
+          min: 1.5,
+          max: 10.0,
+          default: 1000,
```
Comment (jerader) [inline app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: why is the default above the max?

File: app/src/organisms/ChooseRobotSlideout/index.tsx [hunk 3]
```diff
@@ -381,6 +381,13 @@ export function ChooseRobotSlideout(
             tooltipText={runtimeParam.description}
             caption={`${runtimeParam.min}-${runtimeParam.max}`}
             id={id}
+            error={
+              Number.isNaN(value) ||
+              value < runtimeParam.min ||
+              value > runtimeParam.max
+                ? `Value must be between ${runtimeParam.min}-${runtimeParam.max}`
```
Comment (jerader) [inline app/src/organisms/ChooseRobotSlideout/index.tsx]: can you add this to i18n?

**Review by jerader — COMMENTED** [review]
looks good! left a few comments

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx (hunk 1 above)
Comment (ncdiehl11) [reply app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: yes!

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx (hunk 2 above)
Comment (ncdiehl11) [reply app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: mistake from testing. will reset to a reasonable value within range

File: app/src/organisms/ChooseRobotSlideout/index.tsx (hunk 3 above)
Comment (ncdiehl11) [reply app/src/organisms/ChooseRobotSlideout/index.tsx]: yes!

File: app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx [hunk 4]
```diff
@@ -234,6 +234,33 @@ describe('ChooseRobotSlideout', () => {
...
+          description: '70% ethanol volume',
+          type: 'float',
+          suffix: 'mL',
+          min: 1.5,
+          max: 10.0,
+          default: 6.5,
+        },
+      ],
+    })
+    screen.getByText('Value must be between 1.5-10')
```
Comment (koji) [inline app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: question
Don't we need to display `10.0` instead `10`?

Comment (ncdiehl11) [reply app/src/organisms/ChooseRobotSlideout/__tests__/ChooseRobotSlideout.test.tsx]: agreed. From designs, it looks like 1 decimal place for `float`s, and whole integer value for `int`s

File: app/src/organisms/ChooseRobotSlideout/index.tsx [hunk 5]
```diff
@@ -379,8 +379,30 @@ export function ChooseRobotSlideout(
...
             id={id}
+            error={
+              Number.isNaN(value) ||
+              value < runtimeParam.min ||
+              value > runtimeParam.max
+                ? t(`value_out_of_range`, {
+                    min:
+                      runtimeParam.type === 'int'
+                        ? runtimeParam.min
+                        : runtimeParam.min.toFixed(1),
```
Comment (koji) [inline app/src/organisms/ChooseRobotSlideout/index.tsx:398]: 👍

**Review by koji — APPROVED** [review]
lgtm!
thank you for updating the range display part!

**Pull request merged**